packages/andromeda-finance/src/conditional_splitter.rs

```diff
@@ -0,0 +1,282 @@
+}
+
+#[andr_exec]
+#[cw_serde]
+pub enum ExecuteMsg {
+    /// Update the thresholds. Only executable by the contract owner when the contract is not locked.
+    UpdateThresholds { thresholds: Vec<Threshold> },
+    /// Used to lock/unlock the contract allowing the config to be updated.
+    UpdateLock {
+        // Milliseconds from current time
+        lock_time: MillisecondsDuration,
+    },
+    /// Divides any attached funds to the message amongst the recipients list.
+    Send {},
+}
+
+#[andr_query]
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    /// The current config of the Conditional Splitter contract
+    #[returns(GetConditionalSplitterConfigResponse)]
+    GetConditionalSplitterConfig {},
+}
+
+#[cw_serde]
+pub struct GetConditionalSplitterConfigResponse {
+    pub config: ConditionalSplitter,
+}
+
+/// Ensures that a given list of thresholds is valid:
+/// * Percentages of each threshold should not exceed 100
+/// * Each threshold must include at least one recipient
+/// * The number of recipients for each threshold must not exceed 100
+/// * The recipient addresses must be unique for each threshold
+/// * Make sure there are no duplicate min values between the thresholds
+pub fn validate_thresholds(deps: Deps, thresholds: &Vec<Threshold>) -> Result<(), ContractError> {
+    let mut min_value_set = HashSet::new();
+
+    for threshold in thresholds {
+        // Make sure the threshold has recipients
+        ensure!(
+            !threshold.address_percent.is_empty(),
+            ContractError::EmptyRecipientsList {}
+        );
+        // Make sure the threshold's number of recipients doesn't exceed 100
+        ensure!(
+            threshold.address_percent.len() <= 100,
+            ContractError::ReachedRecipientLimit {}
+        );
+
+        let mut total_percent = Decimal::zero();
+        let mut recipient_address_set = HashSet::new();
+
+        for address_percent in &threshold.address_percent {
+            // Check for total percent exceeding 100
+            total_percent = total_percent.checked_add(address_percent.percent)?;
+            ensure!(
+                total_percent <= Decimal::one(),
+                ContractError::AmountExceededHundredPrecent {}
+            );
+
+            // Checks for duplicate and invalid recipients
+            address_percent.recipient.validate(&deps)?;
+            let recipient_address = address_percent.recipient.address.get_raw_address(&deps)?;
+            ensure!(
+                !recipient_address_set.contains(&recipient_address),
+                ContractError::DuplicateRecipient {}
+            );
+            recipient_address_set.insert(recipient_address);
+        }
+
+        // Checks for duplicate minimum values
+        let min_value = threshold.min.u128();
+        ensure!(
+            !min_value_set.contains(&min_value),
+            ContractError::DuplicateThresholds {}
+        );
+
+        min_value_set.insert(min_value);
+    }
```

Comment:
What happens if no threshold is set for `Uint128::zero()`?

Reply:
I think this unit test illustrates that case: `test_execute_send_threshold_not_found`

Reply:
Do we want to require a 0 threshold or is that too limiting?